commcare_connect/opportunity/models.py

```diff
@@ -411,6 +413,7 @@ class UserVisit(XFormBaseModel):
     location = models.CharField(null=True)
     flagged = models.BooleanField(default=False)
     flag_reason = models.JSONField(null=True, blank=True)
+    is_trial = models.BooleanField(default=False)
```

Comment:
I have added this instead of using `flag_reason`, because of two reasons.

1. The way the 'trial' flag is getting used is different to the way the existing `flag_reason`, their purpose seems different. The flags are used to show during a visit-verification, while the `trial` flag needs to be used to exclude the visits that were submitted before start-date
2. It's difficult to query `uservisits` with the way the `flag_reason` is setup now, it can't be queried easily (which needs to happen [here](https://github.com/dimagi/commcare-connect/pull/298/files#diff-0d916b95ee38daa9788eca7b8b3266f406c5c791cab6ff6bda1d5aeefc84d5d2R150) and some other places). We should migrate that to dictionary like below which should make it easier to query 

```
flags = [
    {"name": "duplicate", "description": "A beneficiary with the same identifier already exists"},
    {"name": "duration", "description": "The form was completed too quickly."},
    {"name": "gps", "description": "GPS data is missing"}
]
```

Reply:
That all makes sense over `flag_reason` but I am curious if you considered making this a status option like `OVER_LIMIT`. Both in how it is used and logically, it seems to fit nicely there. Plus it doesn't make a lot of sense to have another status attached to these (a pending trial form, or an over limit trial form). I think that might simplify the receiver logic as well

Reply:
Hmm, how did I miss that attribute. I will change it.

Reply:
(looks like my old comment didn't stick)

Rama was of the opinion that we might care about the visit status of trial forms such as over-limit/duplicate. So `is_trial` becomes necessary.